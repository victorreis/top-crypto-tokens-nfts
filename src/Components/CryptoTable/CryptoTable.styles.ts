import styled from 'styled-components';

export const StyledCryptoTable = styled.table`
  caption-side: top;
  border: none;
  border-spacing: 0;
  caption-side: bottom;

  td,
  th {
    border: none;
  }

  td {
    padding: 1rem 6rem;
  }

  tbody tr {
    :nth-of-type(odd) {
      background-color: ${({ theme }) => theme.colors.main.primary.lighter};
    }
    :hover {
      background-color: ${({ theme }) => theme.colors.main.primary.normal};
    }
  }
`;
