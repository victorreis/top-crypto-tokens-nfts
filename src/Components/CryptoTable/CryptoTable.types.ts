import { TestProps } from '../../Config/Tests/Test.types';
import { CryptoType } from '../../Models/Crypto.types';

export interface RequiredCryptoTableProps {
  /**
   * Crypto data array.
   */
  cryptos: CryptoType[];
}

export interface DefaultCryptoTableProps {}

export interface OptionalCryptoTableProps {}

export type CryptoTableProps = RequiredCryptoTableProps &
  DefaultCryptoTableProps &
  OptionalCryptoTableProps &
  TestProps &
  Omit<React.HTMLAttributes<HTMLSpanElement>, 'children'>;

export type CryptoTableStyleProps = Required<DefaultCryptoTableProps>;
