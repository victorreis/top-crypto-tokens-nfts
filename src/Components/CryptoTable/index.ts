export * from './CryptoTable.types';
export * from './CryptoTable';
