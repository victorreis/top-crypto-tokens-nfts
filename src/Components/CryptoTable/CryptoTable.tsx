/* eslint-disable react/jsx-max-depth */
import { TestProps } from '../../Config/Tests/Test.types';
import { Typography } from '../Typography/Typography';
import { StyledCryptoTable } from './CryptoTable.styles';
import { CryptoTableProps, DefaultCryptoTableProps } from './CryptoTable.types';

export const cryptoTableDefaults: Required<DefaultCryptoTableProps> &
  Required<TestProps> = {
  testID: 'CryptoTable',
};

export const CryptoTable: React.FC<CryptoTableProps> = (props): JSX.Element => {
  const { testID = cryptoTableDefaults.testID, cryptos, ...others } = props;

  const renderCryptoRows = () =>
    cryptos.map(
      ({ id, name, poolPercentage, priceChangePercentage, liquidity }) => (
        <tr key={id}>
          <td>
            <Typography>{name}</Typography>
          </td>
          <td>
            <Typography>{poolPercentage}</Typography>
          </td>
          <td>
            <Typography>{priceChangePercentage}</Typography>
          </td>
          <td>
            <Typography>{liquidity}</Typography>
          </td>
        </tr>
      )
    );

  return (
    <div data-testid={testID} {...others}>
      <StyledCryptoTable>
        <thead>
          <tr>
            <th>
              <Typography variant="subtitle1">Tokens</Typography>
            </th>
            <th>
              <Typography variant="subtitle1">Pool %</Typography>
            </th>
            <th>
              <Typography variant="subtitle1">Price Change %</Typography>
            </th>
            <th>
              <Typography variant="subtitle1">Liquidity</Typography>
            </th>
          </tr>
        </thead>
        <tbody>{renderCryptoRows()}</tbody>
      </StyledCryptoTable>
    </div>
  );
};
