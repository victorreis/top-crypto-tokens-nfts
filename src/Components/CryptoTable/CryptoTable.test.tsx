import {
  renderJestDomCreator,
  renderRTRCreator,
  screen,
} from '../../Config/Tests/GlobalSetup.config';
import { CryptoType } from '../../Models/Crypto.types';
import { CryptoTable, cryptoTableDefaults } from './CryptoTable';
import {
  RequiredCryptoTableProps,
  CryptoTableProps,
} from './CryptoTable.types';

describe('cryptoTable component tests', () => {
  const cryptos: CryptoType[] = [
    {
      id: '',
      name: '',
      poolPercentage: '',
      priceChangePercentage: '',
      liquidity: 10000.1,
    },
  ];

  const requiredProps: RequiredCryptoTableProps = {
    cryptos,
  };

  const setup = (props?: CryptoTableProps) => {
    const renderRTR = () =>
      renderRTRCreator<CryptoTableProps>(CryptoTable, {
        ...requiredProps,
        ...props,
      });
    const renderJestDom = () =>
      renderJestDomCreator<CryptoTableProps>(CryptoTable, {
        ...requiredProps,
        ...props,
      });

    return { renderRTR, renderJestDom };
  };

  describe('behavior tests', () => {
    it(`should render the component`, () => {
      expect.assertions(1);
      setup().renderJestDom();
      const testInstance = screen.getByTestId(cryptoTableDefaults.testID);

      expect(testInstance).toBeTruthy();
    });
  });

  describe('style tests', () => {});

  describe('snapshot tests', () => {
    it(`should render correctly`, () => {
      expect.assertions(1);
      const generatedJson = setup().renderRTR().toJSON();

      expect(generatedJson).toMatchSnapshot();
    });
  });
});
