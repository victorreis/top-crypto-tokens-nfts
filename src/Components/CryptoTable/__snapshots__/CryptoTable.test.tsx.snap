// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cryptoTable component tests snapshot tests should render correctly 1`] = `
<div
  data-testid="CryptoTable"
>
  <table
    className="StyledCryptoTable-react-typescript-template__sc-cy4piq-0 ibsjOI"
  >
    <thead>
      <tr>
        <th>
          <h3
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 bCcLTj"
            data-testid="Typography"
          >
            Tokens
          </h3>
        </th>
        <th>
          <h3
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 bCcLTj"
            data-testid="Typography"
          >
            Pool %
          </h3>
        </th>
        <th>
          <h3
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 bCcLTj"
            data-testid="Typography"
          >
            Price Change %
          </h3>
        </th>
        <th>
          <h3
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 bCcLTj"
            data-testid="Typography"
          >
            Liquidity
          </h3>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 cBRKtu"
            data-testid="Typography"
          />
        </td>
        <td>
          <span
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 cBRKtu"
            data-testid="Typography"
          />
        </td>
        <td>
          <span
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 cBRKtu"
            data-testid="Typography"
          />
        </td>
        <td>
          <span
            className="TypographyContainer-react-typescript-template__sc-foyag6-0 cBRKtu"
            data-testid="Typography"
          >
            10000.1
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
